components/ListOfItemsCollection.vue

```diff
@@ -217,87 +222,56 @@ useHead({
         />
         <DividerWayFinder />
 
-        <span class="search-filters">
+        <span
+          v-if="!isLoading"
+          class="search-filters"
+        >
           <!-- Filter by -->
-          <div
-            v-for="(filter, index) in searchFilters"
-            :key="index"
-          >
-            <label
-              v-if="filter.label"
-              :for="filter.label"
-              class="select-label"
-            >
-              {{ filter.label }}
-            </label>
-            <select
-              v-if="filter.label"
-              :id="filter.label"
-              v-model="userFilterSelectionValue"
-              :name="filter.label"
-              class="select-input"
-              @change="(e) => {
-                if (e.target.value === '(none selected)') {
-                  router.push({
-                    path: route.path,
-                    query: {
-                      sort: userSortSelection,
-                      page: route.query.page,
-                    }
-                  })
-                } else {
-                  router.push({
-                    path: route.path,
-                    query: {
-                      filters: [fieldNamefromLabel[filter.label]] + ':(' + e.target.value + ')',
-                      sort: userSortSelection,
-                      page: route.query.page,
-                    }
-                  })
-                }
-              }"
-            >
-              <option
-                disabled
-                value=""
-              >
-                -- Select an option --
-              </option>
-              <option
-                v-for="option in filter.options"
-                :key="option.key"
-                :value="option.value"
-              >
-                {{ option.label }}
-              </option>
-            </select>
-          </div>
-          <!-- Sort Order -->
-          <div v-if="!isLoading">
-            <label
-              for="sortorder"
-              class="sort-label"
-            >Sort order</label>
-            <select
-              id="sortorder"
-              v-model="userSortSelection"
-              name="sortorder"
-              @change="(e) => {
+          <DropdownSingleSelect
+            v-model:selected-filters="selectedFilters"
+            :label="searchFilters[0].label"
+            :options="searchFilters[0].options"
+            :field-name="fieldNamefromLabel[searchFilters[0].label]"
+            @update-display="(newFilter) => {
+              const newFilterValue = Object.values(newFilter)[0]
```

Comment:
@farosFreed Can both the `update-display` event logic be moved outside of the template?